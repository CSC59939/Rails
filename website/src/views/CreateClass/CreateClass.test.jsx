import React from 'react';
import renderer from 'react-test-renderer';
import {
  configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CreateClass } from '..';

configure({ adapter: new Adapter() });

describe('CreateClass', () => {
  it('Render CreateClass without any eror', () => { // eslint-disable-line no-undef
    expect(renderer.create(<CreateClass />).toJSON()).toMatchSnapshot();
  });
});
