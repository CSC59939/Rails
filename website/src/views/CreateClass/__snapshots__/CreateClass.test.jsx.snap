// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateClass Render CreateClass without any eror 1`] = `
<div
  className="create"
>
  <h1
    className="title"
  >
    Rails
  </h1>
  <div
    className="ant-card createcard ant-card-bordered"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Create Class
        </div>
        <div
          className="ant-card-extra"
        >
          <a
            href="/join/class"
          >
            Already have a class code? Join Class
          </a>
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <form
        className="ant-form ant-form-horizontal regiform"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <div
                  className="ant-select ant-select-enabled"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <div
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup="true"
                    className="ant-select-selection
            ant-select-selection--single"
                    onKeyDown={[Function]}
                    role="combobox"
                    tabIndex={0}
                  >
                    <div
                      className="ant-select-selection__rendered"
                    >
                      <div
                        className="ant-select-selection__placeholder"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        style={
                          Object {
                            "WebkitUserSelect": "none",
                            "display": "block",
                            "userSelect": "none",
                          }
                        }
                        unselectable="on"
                      >
                        University/College
                      </div>
                      <div
                        className="ant-select-search ant-select-search--inline"
                        style={
                          Object {
                            "display": "none",
                          }
                        }
                      >
                        <div
                          className="ant-select-search__field__wrap"
                        >
                          <input
                            autoComplete="off"
                            className="ant-select-search__field"
                            onChange={[Function]}
                            onKeyDown={[Function]}
                            value=""
                          />
                          <span
                            className="ant-select-search__field__mirror"
                          >
                            
                             
                          </span>
                        </div>
                      </div>
                    </div>
                    <span
                      className="ant-select-arrow"
                      onClick={[Function]}
                      style={
                        Object {
                          "WebkitUserSelect": "none",
                          "userSelect": "none",
                        }
                      }
                      unselectable="on"
                    >
                      <i
                        className="anticon anticon-down ant-select-arrow-icon"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="down"
                          fill="currentColor"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </i>
                    </span>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <input
                  className="ant-input"
                  disabled={false}
                  onKeyDown={[Function]}
                  placeholder="Class Name"
                  type="text"
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <input
                  className="ant-input"
                  disabled={false}
                  onKeyDown={[Function]}
                  placeholder="Section code"
                  type="text"
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <span>
                  Meeting Dates 
                </span>
                <span
                  className="ant-calendar-picker"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={Object {}}
                  tabIndex={0}
                >
                  <span
                    className="ant-calendar-picker-input ant-input"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                  >
                    <input
                      className="ant-calendar-range-picker-input"
                      placeholder="Start date"
                      readOnly={true}
                      tabIndex={-1}
                      value=""
                    />
                    <span
                      className="ant-calendar-range-picker-separator"
                    >
                       ~ 
                    </span>
                    <input
                      className="ant-calendar-range-picker-input"
                      placeholder="End date"
                      readOnly={true}
                      tabIndex={-1}
                      value=""
                    />
                    <i
                      className="anticon anticon-calendar ant-calendar-picker-icon"
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="calendar"
                        fill="currentColor"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                        />
                      </svg>
                    </i>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <span>
                  Days & Times 
                </span>
                <div
                  className="ant-checkbox-group"
                >
                  <label
                    className="ant-checkbox-group-item ant-checkbox-wrapper"
                  >
                    <span
                      className="ant-checkbox"
                      style={Object {}}
                    >
                      <input
                        checked={false}
                        className="ant-checkbox-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                        value="Mon"
                      />
                      <span
                        className="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      Mon
                    </span>
                  </label>
                  <label
                    className="ant-checkbox-group-item ant-checkbox-wrapper"
                  >
                    <span
                      className="ant-checkbox"
                      style={Object {}}
                    >
                      <input
                        checked={false}
                        className="ant-checkbox-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                        value="Tue"
                      />
                      <span
                        className="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      Tue
                    </span>
                  </label>
                  <label
                    className="ant-checkbox-group-item ant-checkbox-wrapper"
                  >
                    <span
                      className="ant-checkbox"
                      style={Object {}}
                    >
                      <input
                        checked={false}
                        className="ant-checkbox-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                        value="Wed"
                      />
                      <span
                        className="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      Wed
                    </span>
                  </label>
                  <label
                    className="ant-checkbox-group-item ant-checkbox-wrapper"
                  >
                    <span
                      className="ant-checkbox"
                      style={Object {}}
                    >
                      <input
                        checked={false}
                        className="ant-checkbox-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                        value="Thu"
                      />
                      <span
                        className="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      Thu
                    </span>
                  </label>
                  <label
                    className="ant-checkbox-group-item ant-checkbox-wrapper"
                  >
                    <span
                      className="ant-checkbox"
                      style={Object {}}
                    >
                      <input
                        checked={false}
                        className="ant-checkbox-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                        value="Fri"
                      />
                      <span
                        className="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      Fri
                    </span>
                  </label>
                  <label
                    className="ant-checkbox-group-item ant-checkbox-wrapper"
                  >
                    <span
                      className="ant-checkbox"
                      style={Object {}}
                    >
                      <input
                        checked={false}
                        className="ant-checkbox-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                        value="Sat"
                      />
                      <span
                        className="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      Sat
                    </span>
                  </label>
                  <label
                    className="ant-checkbox-group-item ant-checkbox-wrapper"
                  >
                    <span
                      className="ant-checkbox"
                      style={Object {}}
                    >
                      <input
                        checked={false}
                        className="ant-checkbox-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                        value="Sun"
                      />
                      <span
                        className="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      Sun
                    </span>
                  </label>
                </div>
                <span
                  className="ant-time-picker "
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                  style={Object {}}
                >
                  <input
                    className="ant-time-picker-input"
                    disabled={false}
                    id=""
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Select time"
                    readOnly={false}
                    type="text"
                    value=""
                  />
                  <span
                    className="ant-time-picker-icon"
                  >
                    <i
                      className="anticon anticon-clock-circle ant-time-picker-clock-icon"
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="clock-circle"
                        fill="currentColor"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                        <path
                          d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                        />
                      </svg>
                    </i>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <i
                  className="anticon anticon-question-circle-o"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="question-circle"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    />
                    <path
                      d="M623.6 316.7C593.6 290.4 554 276 512 276s-81.6 14.5-111.6 40.7C369.2 344 352 380.7 352 420v7.6c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V420c0-44.1 43.1-80 96-80s96 35.9 96 80c0 31.1-22 59.6-56.1 72.7-21.2 8.1-39.2 22.3-52.1 40.9-13.1 19-19.9 41.8-19.9 64.9V620c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-22.7a48.3 48.3 0 0 1 30.9-44.8c59-22.7 97.1-74.7 97.1-132.5.1-39.3-17.1-76-48.3-103.3zM472 732a40 40 0 1 0 80 0 40 40 0 1 0-80 0z"
                    />
                  </svg>
                </i>
                <input
                  className="ant-input"
                  disabled={false}
                  onKeyDown={[Function]}
                  placeholder="Email"
                  type="text"
                />
              </span>
            </div>
          </div>
        </div>
        <div
          align="center"
          className="registerButton"
        >
          <button
            className="ant-btn ant-btn-primary"
            margin="auto"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Create
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
